### DISCUSSION TITLE
Do I need to do anything special to make Verify work with SpecFlow?

### DISCUSSION THREAD

**Discussion by baynezy** [VerifyTests/Verify#513] [General]
I am trying to use Verify with SpecFlow and I have added the `[UsesVerify]` attribute to the step definitions class. Howver, when I run the feature it complains that `Expected to find a `[UsesVerify]` on test class. File: DocumentStepDefinitions.cs` `DocumentStepDefinitions.cs`` currently does have the `[UsesVerify]` attribute so I am not sure what to do next.

**Comment by SimonCropp** [comment]
can you share a full test class that reproduces this issue?

**Reply by baynezy** [reply]
I am struggling to recreate my exact issue in a separate repo. So it is likely a something I have done wrong. However, on doing that I have an issue where Verify runs but doesn't really test.

https://github.com/baynezy/SpecFlowVerifyIssue

If you check that out and run `dotnet test` it has one passing test. My assumption is that [this code](https://github.com/baynezy/SpecFlowVerifyIssue/blob/master/test/Example.Spec/Steps/HtmlGeneratorStepDefinitions.cs#L31-L35) would mean that is would create a file on first run, but it creates nothing.

What am I doing wrong?

**Reply by kaylumah** [reply]
@baynezy did you manage to get it working?

**Reply by kaylumah** [reply]
My assumption is that we need to get the attribute in the generated code not the definitions
Raised an issue https://github.com/SpecFlowOSS/SpecFlow/issues/2633 on how to do this in xunit

**Reply by baynezy** [reply]
@kaylumah I didn't manage to get it working, no. 

It turns out that it is a limitation of SpecFlow https://github.com/SpecFlowOSS/SpecFlow/issues/2601

It's resolvable, but I've not had time to progress it.

**Reply by kaylumah** [reply]
@baynezy the draft Pr looks promesing
Just confirmed that the proposed workaround (using partial class) is working

**Reply by SimonCropp** [reply]
yeah, sorry, not much i can do from my side

**Reply by SimonCropp** [reply]
as soon as specflow adds support for that attribute, i will add docs around working with specflow

**Reply by SabotageAndi** [reply]
I am working on it: https://github.com/SpecFlowOSS/SpecFlow/pull/2602

**Comment by SabotageAndi** [comment]
So, Verify support for SpecFlow is available.
Needed NuGet package: https://www.nuget.org/packages/SpecFlow.Verify/4.0.16-beta
Example: https://github.com/SpecFlowOSS/SpecFlow-Examples/tree/master/Verify/VerifyExample
